### ISSUE TITLE
Can't resolve 'react-native' when using @walletconnect/ethereum-provider

### ISSUE THREAD

**Issue by alxlacayo** [WalletConnect/walletconnect-monorepo#2293]
Trying to obtain the ethereum provider object via EthereumProvider in an Next.js but am getting the following error:

```
./node_modules/@walletconnect/universal-provider/dist/index.es.js
Module not found: Can't resolve 'react-native' in '/Users/*/Desktop/frontend/node_modules/@walletconnect/universal-provider/dist'
```

```
import { EthereumProvider } from '@walletconnect/ethereum-provider'

const provider = await EthereumProvider.init({
  projectId: "XXXXXXX",
  chains: [1],
  showQrModal: true
})
```

I am following the docs [here](https://docs.walletconnect.com/2.0/web/providers/ethereum). Do I really need to install react-native for this?

**Comment by josiah-ejs** [comment]
I encounter same error when building React Web App. Issue remains even after react-native installation. Issue was temporary resolved by falling back to version 2.7.0 [npm install @walletconnect/ethereum-provider@2.7.0]. Please post here if there is a better solution instead of the falling back.

**Comment by opeoyeleke** [comment]
Having the same issue since yesterday. trying the fix by @josiah-ejs hasnt worked for me so far

**Comment by opeoyeleke** [comment]
Editing yarn.lock and changing the versions of all "@walletconnect/ethereum-provider" instances from 2.7.2 to 2.7.0 worked for me

**Comment by tempest-sol** [comment]
This resolved the issue temporarily for me, put in 'package.json'

```json
  "resolutions": {
    "@walletconnect/ethereum-provider": "2.7.0"
  }

**Comment by paintoshi** [comment]
I had the same issue but when using the web3modal and specifically wagmi@0.12.13

From yarn.lock:
![image](https://user-images.githubusercontent.com/88572775/237033908-969e4e2e-50e1-46d5-b7ca-ecab5b32293a.png)

Downgrading to wagmi@0.12.12 solved the issue (which uses ethereum-provider 2.7.0) . The resolution trick by @tempest-sol (followed by a yarn install) also worked but with a compatibility warning instead.

**Comment by ganchoradkov** [comment]
Hey. we've just released `2.7.4` and the issue should be fixed

**Comment by hammeiam** [comment]
@ganchoradkov Does this fix also exist for the `0.12.15` tag? Getting the same issue there
